FrontEnd/App.js

```diff
@@ -18,21 +18,30 @@ const reducer = combineReducers({
 
 const store = createStore(reducer, install());
 
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-  }
+function App(props) {
+    props.loadUserData(localStorage.getItem("userId"), localStorage.getItem("role"), localStorage.getItem("request_token"))
+    const [update, setUpdate] = useState(false);
 
-  render() {
-    const appropriateNavigation = this.props.role === "student" ? <SchuelerHauptmenue /> : <LehrerNavigation />;
-    const loginScreenOrNavigation = this.props.loggedIn ? appropriateNavigation : <LoginScreen />;
+    function handleUpdate() {
+        setUpdate(!update);
+    }
+
+    const appropriateNavigation = localStorage.getItem("role") === "student" ? <SchuelerHauptmenue handleLogout={handleUpdate} />: <LehrerNavigation handleLogout={handleUpdate} />;
```

Comment:
We don't do that here. Genau dafür ist Redux da. Dass du nicht irgendwelche Funktionen als Props quer durch die App reichst.